UnitTests/GitExtUtils/GitArgumentBuilderTests.cs

```diff
@@ -168,5 +169,29 @@ public void Throws_for_invalid_command_strings()
             Assert.Throws<ArgumentException>(() => new GitArgumentBuilder(" a banana "));
             Assert.Throws<ArgumentException>(() => new GitArgumentBuilder("!£$%^&*("));
         }
+
+        [Test]
+        public void Length()
+        {
+            var builder = new GitArgumentBuilder("foo");
+            builder.GetLength().Should().Be(/* 'foo ' */4);
+
+            builder.Add("test");
+            builder.GetLength().Should().Be(/* 'foo test'*/8);
+
+            var args = "Lorem ipsum dolor sit amet, solet soleat option mel no.";
+            var expectedLength = args.Length;
+            builder.AddRange(args.Split(' '));
+            builder.GetLength().Should().Be(expectedLength + /* 'foo test ' */9);
+
+            builder = new GitArgumentBuilder("for-each-ref", gitOptions: "--no-optional-locks")
+            {
+                "--sort=-committerdate",
+                "--format=\"%(objectname) %(refname)\"",
+                "refs/"
+            };
+            var rendered = builder.ToString();
+            builder.GetLength().Should().Be(rendered.Length);
```

Comment:
Nit: I would call `GetLength` before `ToString` because the state of `builder` could change in theory. (C# should have `const` methods.)